src/libs/image.c

```diff
@@ -85,11 +85,14 @@ uint32_t container(dt_lib_module_t *self)
 }
 
 /** merges all the selected images into a single group.  if there is
- * an expanded group, then they will be joined there, otherwise a new
+ * an expanded group and grouping is on, then they will be joined there, otherwise a new
  * one will be created. */
 static void _group_helper_function(void)
 {
-  dt_imgid_t new_group_id = darktable.gui->expanded_group_id;
+  dt_imgid_t new_group_id = NO_IMGID;
```

Comment:
Maybe rewrite as:

```
const dt_imgid_t new_group_id = darktable.gui->grouping
   ? darktable.gui->expanded_group_id;
   : NO_IMGID;
```

Reply:
Can't be a `const` as new_group_id gets used to create a new group a few lines later.  When it's set to NO_IMGID it takes the first selected image id and assigns it to new_image_id for use in building the group.